src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -64,12 +64,20 @@ public bool Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnl
 				}
 
 				foreach (var value in argumentValues[0]) {
-					if (value is RuntimeTypeHandleValue typeHandle)
-						AddReturnValue (new SystemTypeValue (typeHandle.RepresentedType));
-					else if (value is RuntimeTypeHandleForGenericParameterValue typeHandleForGenericParameter)
-						AddReturnValue (GetGenericParameterValue (typeHandleForGenericParameter.GenericParameter));
-					else
-						AddReturnValue (GetMethodReturnValue (calledMethod, returnValueDynamicallyAccessedMemberTypes));
+					AddReturnValue (value switch {
+						NullableRuntimeSystemTypeHandleValue nullableSystemType
+							=> new NullableSystemTypeValue (nullableSystemType.NullableType, nullableSystemType.UnderlyingTypeValue),
+						NullableRuntimeTypeWithDamHandleValue nullableDamType when nullableDamType.UnderlyingTypeValue is RuntimeTypeHandleForGenericParameterValue underlyingGenericParameter
```

Comment:
Does this need a corresponding case in `Type_get_TypeHandle` that turns a `NullableValueWithDynamicallyAccessedMembers` of `GenericParameterValue` back into a `NullableRuntimeTypeWithDamHandleValue` of `RuntimeTypeHandleForGenericParameterValue`? I am wondering why there's an asymmetry between the intrinsics that go from TypeHandle -> Type vs Type -> TypeHandle.

Reply:
Yes, there was a hole here, thank you!